src/java.base/share/classes/sun/security/ssl/SSLSessionImpl.java

```diff
@@ -1456,6 +1473,267 @@ public List<SNIServerName> getRequestedServerNames() {
+            // RFC 7505 using PRF-based calcs.
+            //     TLS 1/1.1/1.2 (RFCs 2246/4346/5246) or
+            //     DTLS 1.0/1.2 (RFCs 4347/6347)
+
+            // Note:  In RFC 7627:
+            //
+            //   If a client or server chooses to continue with a full handshake
+            //   without the extended master secret extension ... they MUST NOT
+            //   export any key material based on the new master secret for any
+            //   subsequent application-level authentication ... it MUST
+            //   disable [RFC5705] ...
+
+            if (!useExtendedMasterSecret) {
+                throw new SSLKeyException(
+                        "Exporters require extended master secrets");
+            }
+
+            // Check for a "disambiguating label string" (i.e. non-empty).
+            // Don't see a max length restriction.
+            if (label.isEmpty()) {
+                throw new IllegalArgumentException(
+                        "label length outside range");
+            }
+
+            // Unlikely, but check if randoms were not captured.
+            if (clientRandom == null || serverRandom == null) {
+                throw new RuntimeException("Random nonces not captured");
+            }
+
+            // context length must fit in 2 unsigned bytes.
+            if ((context != null) && (context.length > 0xFFFF)) {
+                throw new IllegalArgumentException(
+                        "Only 16-bit context lengths supported");
+            }
+
+            // Perform RFC 5705 calculations using the internal SunJCE PRF.
+            String prfAlg;
+            HashAlg hashAlg;
+            if (protocolVersion == TLS12) {
+                prfAlg = "SunTls12Prf";
+                hashAlg = cipherSuite.hashAlg;
+            } else {  //  all other cases
+                prfAlg = "SunTlsPrf";
+                hashAlg = H_NONE;
+            }
+
+            // Make a seed with randoms and optional context
+            // Note that if context is null, it is omitted from the calc
+            byte[] clientRandomBytes = clientRandom.randomBytes;
+            byte[] serverRandomBytes = serverRandom.randomBytes;
+            byte[] seed = new byte[
+                    clientRandomBytes.length + serverRandomBytes.length +
+                            ((context != null) ? (2 + context.length) : 0)];
+
+            int pos = 0;
+            System.arraycopy(
+                    clientRandomBytes, 0, seed, pos, clientRandomBytes.length);
+            pos += clientRandomBytes.length;
+            System.arraycopy(
+                    serverRandomBytes, 0, seed, pos, serverRandomBytes.length);
+            pos += serverRandomBytes.length;
+            if (context != null) {
+                // RFC 5705, "If no context is provided, ..."
+                seed[pos++] = (byte) ((context.length >> 8) & 0xFF);
+                seed[pos++] = (byte) ((context.length) & 0xFF);
```

Comment:
Because you're doing a primitive narrowing conversion down to a byte I don't think you need the 0xFF masking.

Reply:
No, not required.  `java.lang.Integer.byteValue()` does exactly this.  

But I was thinking to leave so the intent was clear.  We talked, and will leave for now.